public/app/features/dashboard/api/v0.ts

```diff
@@ -90,32 +91,46 @@ export class K8sDashboardAPI implements DashboardAPI<DashboardDTO> {
   }
 
   async getDashboardDTO(uid: string) {
-    const dash = await this.client.subresource<DashboardWithAccessInfo<DashboardDataDTO>>(uid, 'dto');
-
-    const result: DashboardDTO = {
-      meta: {
-        ...dash.access,
-        isNew: false,
-        isFolder: false,
-        uid: dash.metadata.name,
-        k8s: dash.metadata,
-        version: parseInt(dash.metadata.resourceVersion, 10),
-      },
-      dashboard: dash.spec,
-    };
+    try {
+      const dash = await this.client.subresource<DashboardWithAccessInfo<DashboardDataDTO>>(uid, 'dto');
+
+      const result: DashboardDTO = {
+        meta: {
+          ...dash.access,
+          isNew: false,
+          isFolder: false,
+          uid: dash.metadata.name,
+          k8s: dash.metadata,
+          version: parseInt(dash.metadata.resourceVersion, 10),
+        },
+        dashboard: dash.spec,
+      };
 
-    if (dash.metadata.annotations?.[AnnoKeyFolder]) {
-      try {
-        const folder = await backendSrv.getFolderByUid(dash.metadata.annotations[AnnoKeyFolder]);
-        result.meta.folderTitle = folder.title;
-        result.meta.folderUrl = folder.url;
-        result.meta.folderUid = folder.uid;
-        result.meta.folderId = folder.id;
-      } catch (e) {
-        console.error('Failed to load a folder', e);
+      if (dash.metadata.annotations?.[AnnoKeyFolder]) {
+        try {
+          const folder = await backendSrv.getFolderByUid(dash.metadata.annotations[AnnoKeyFolder]);
+          result.meta.folderTitle = folder.title;
+          result.meta.folderUrl = folder.url;
+          result.meta.folderUid = folder.uid;
+          result.meta.folderId = folder.id;
+        } catch (e) {
+          console.error('Failed to load a folder', e);
+        }
+      }
+
+      return result;
+    } catch (e) {
```

Comment:
Is this catch going to be reached if we catch it here as well?https://github.com/grafana/grafana/pull/98950/files#diff-2b81dbfdccf7fae22386553e8f93600dc2786072e77683d809da06c8768331ceR117

Is it on purpose to not throw a folder not found error and use console.error instead? This means we could return a Dashboard DTO without folder info.

Reply:
good catch! fixed in 933ec5d26fc38fdc7f71f55daa06e70efb785591